Missing checklist information in API-docs

### background
This issue is in response to the suggestion of @Alys in the Comrade guild chat.


### Description
*Summary:* Checklist information is missing in the habitica [API-docs](https://habitica.com/apidoc/)


#### 1. CreateUserTasks endpoint
The API-docs for <https://habitica.com/apidoc/#api-Task-CreateUserTasks> are missing the `checklist` body parameter.
The wiki does give a good example here [source](https://habitica.fandom.com/wiki/Application_Programming_Interface):
```bash
$ curl -X POST \
    # ...
    -d '{
    "checklist": [
            { "text": "1st checklist item", "completed": false },
            { "text": "2nd checklist item", "completed": false }
        ],
```

![apidocs](https://user-images.githubusercontent.com/31448155/129471651-c88036a3-02a5-4ae5-9aa2-71d989f51191.png)


#### 2. AddChecklistItem and UpdateChecklistItem endpoint
As I was writing this down, I noticed that other endpoints are also missing information that is implicitly mentioned in the wiki.
For example, <https://habitica.com/apidoc/#api-Task-AddChecklistItem> does not make mention of `<completed>` body parameter. It is also unclear if the <https://habitica.com/apidoc/#api-Task-UpdateChecklistItem> supports the `<completed>` body.

Thanks @Alys. I managed to run habitica. Which was a breeze with the tips listed over here <https://habitica.fandom.com/wiki/Setting_up_Habitica_Locally_on_Docker>. I basically only had to fork the project, run `cp config.json{.example,}` and run `docker-compose -f docker-compose.dev.yml up -d` and I was good to go. 

The output looks as I want it to be on <localhost:3000/apidoc>. e.g.:
![apidocs-update](https://user-images.githubusercontent.com/31448155/131212624-81b7ad6a-49ca-4411-bfd0-4d1b55964fd7.png)

However, the commit I was working from was already getting a few days old. 
I've checked it out Wednesday the 18th of August (10 days ago) but I didn't work on it till this morning: `ca185f019839555d9c9dfa67c5a283e83b53b838`.
So I've updated the code to be up to date with `develop` (commit: `5b346690a9cde8a09afaa3ec9770da46e918e8e6`).

Unfortunately, the docker container `habitica-server` dies consistently after a few seconds with this version.

I think that `docker-compose logs server` gives us the error:
```
Attaching to habitica_server_1
server_1  | 
server_1  | > habitica@4.203.1 start /usr/src/habitica
server_1  | > gulp nodemon
server_1  | 
server_1  | [09:05:19] Using gulpfile /usr/src/habitica/gulpfile.js
server_1  | [09:05:19] Starting 'nodemon'...
server_1  | [09:05:19] Finished 'nodemon' after 56 ms
server_1  | [09:05:19] [nodemon] 2.0.4
server_1  | [09:05:19] [nodemon] to restart at any time, enter `rs`
server_1  | [09:05:19] [nodemon] watching path(s): *.*
server_1  | [09:05:19] [nodemon] watching extensions: js,mjs,json
server_1  | [09:05:19] [nodemon] starting `node ./website/server/index.js`
server_1  | 2021-08-28T09:05:28.591Z - info Express server listening on port 3000
server_1  | 2021-08-28T09:05:28.613Z - info Connected with Mongoose.
server_1  | npm ERR! code ELIFECYCLE
server_1  | npm ERR! errno 1
server_1  | npm ERR! habitica@4.203.1 start: `gulp nodemon`
server_1  | npm ERR! Exit status 1
server_1  | npm ERR! 
server_1  | npm ERR! Failed at the habitica@4.203.1 start script.
server_1  | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
server_1  | 
server_1  | npm ERR! A complete log of this run can be found in:
server_1  | npm ERR!     /root/.npm/_logs/2021-08-28T09_05_37_346Z-debug.log
```
@Alys Do you know what happened in the past few days that could cause `npm/nodemon/gulp` to fail inside the `habitica-server` container? 
Getting habitica to run 10 days ago was really surprisingly easy so it might have to do with version bumps performed by the @dependabot ?

Note: I've also attached the results of the client (`docker-compose logs --no-color client > docker_client_logs.txt`) but I see no errors there. 
[docker_client_logs.txt](https://github.com/HabitRPG/habitica/files/7070466/docker_client_logs.txt)

Hey @SabreCat Great to meet you :smiley:. Building with the most recent `develop` version, (`5b346690a9cde8a09afaa3ec9770da46e918e8e6`). with the commit message `fix(storybook): temporarily disable story`, gives me the error listed above. 

To be sure, I also asked `docker-compose` to clean up thoroughly:
```bash
docker-compose --file docker-compose.dev.yml up --force-recreate --always-recreate-deps --remove-orphans --detach
```

Unfortunately, the response is the same:
```
$ docker logs habitica_server_1 

> habitica@4.203.1 start /usr/src/habitica
> gulp nodemon

[06:29:39] Using gulpfile /usr/src/habitica/gulpfile.js
[06:29:39] Starting 'nodemon'...
[06:29:40] Finished 'nodemon' after 47 ms
[06:29:40] [nodemon] 2.0.4
[06:29:40] [nodemon] to restart at any time, enter `rs`
[06:29:40] [nodemon] watching path(s): *.*
[06:29:40] [nodemon] watching extensions: js,mjs,json
[06:29:40] [nodemon] starting `node ./website/server/index.js`
2021-08-29T06:29:45.815Z - info Express server listening on port 3000
2021-08-29T06:29:45.836Z - info Connected with Mongoose.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! habitica@4.203.1 start: `gulp nodemon`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the habitica@4.203.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I do notice the `npm ERR` message stating the `habitica@4.203.1 start script` part.      
Besides checking out the most recent commit with git, do I also need to modify the `package.json`'s `"version":4.203.1"` " to run a more recent version?

PS: I also tried `release` but the errors are identical.

I tagged the most recent version of habitica on my `melvio/habitica` repo with `4.203.2`.   
I incremented the package.json version string to `"version": "4.203.2",`. (just locally, I did not push)

I ran docker-compose with the `--build` flag this time:
```
docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d --force-recreate --build
```

Unfortunately, the error message remains the same:
```
> habitica@4.203.2 start /usr/src/habitica
> gulp nodemon

[07:10:21] Using gulpfile /usr/src/habitica/gulpfile.js
[07:10:21] Starting 'nodemon'...
[07:10:21] Finished 'nodemon' after 53 ms
[07:10:21] [nodemon] 2.0.4
[07:10:21] [nodemon] to restart at any time, enter `rs`
[07:10:21] [nodemon] watching path(s): *.*
[07:10:21] [nodemon] watching extensions: js,mjs,json
[07:10:21] [nodemon] starting `node ./website/server/index.js`
2021-08-29T07:10:27.222Z - info Express server listening on port 3000
2021-08-29T07:10:27.255Z - info Connected with Mongoose.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! habitica@4.203.2 start: `gulp nodemon`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the habitica@4.203.2 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

--- a/website/server/controllers/api-v3/tasks.js
+++ b/website/server/controllers/api-v3/tasks.js
@@ -61,11 +61,14 @@ const api = {};
  * @apiParam (Body) {String="str","int","per","con"} [attribute] User's attribute to use,
  *                                                               options are: "str", "int",
  *                                                               "per", "con"
+ * @apiParam (Body) {Array} [checklist] An array of checklist items. For example,
+ *                                      [{"text":"buy tools", "completed":true},
+ *                                       {"text":"build shed", "completed":false}]
  * @apiParam (Body) {Boolean} [collapseChecklist=false] Determines if a checklist will be displayed
  * @apiParam (Body) {String} [notes] Extra notes
  * @apiParam (Body) {Date} [date] Due date to be shown in task list. Only valid for type "todo."
  * @apiParam (Body) {Number="0.1","1","1.5","2"} [priority=1] Difficulty, options are 0.1, 1,
- *                                                            1.5, 2; eqivalent of Trivial,
+ *                                                            1.5, 2; equivalent of Trivial,
  *                                                            Easy, Medium, Hard.
  * @apiParam (Body) {String[]} [reminders] Array of reminders, each an object that must
  *                                         include: a UUID, startDate and time.
@@ -110,6 +113,7 @@ const api = {};
  *       "alias":"hab-api-tasks",
  *       "notes":"Update the tasks api on GitHub",
  *       "tags":["ed427623-9a69-4aac-9852-13deb9c190c3"],
+ *       "checklist":[{"text":"read wiki","completed":true},{"text":"write code"}],
  *       "priority":2
  *     }
  *
@@ -150,7 +154,16 @@ const api = {};
  *         "createdAt": "2017-01-12T02:11:02.876Z",
  *         "updatedAt": "2017-01-12T02:11:02.876Z",
  *         "checklist": [
- *
+ *            {
+ *              "completed": true,
+ *              "text": "read wiki",
+ *              "id": "91edadda-fb62-4e6e-b110-aff26f936678"
+ *            },
+ *            {
+ *              "completed": false,
+ *              "text": "write code",
+ *              "id": "d1ddad50-ab22-49c4-8261-9996ae337b6a"
+ *            }
  *         ],
  *         "collapseChecklist": false,
  *         "completed": false,
@@ -236,7 +249,7 @@ api.createUserTasks = {
  * @apiParam (Body) {String} [notes] Extra notes
  * @apiParam (Body) {Date} [date] Due date to be shown in task list. Only valid for type "todo."
  * @apiParam (Body) {Number="0.1","1","1.5","2"} [priority=1] Difficulty, options are 0.1, 1,
- *                                                            1.5, 2; eqivalent of Trivial,
+ *                                                            1.5, 2; equivalent of Trivial,
  *                                                            Easy, Medium, Hard.
  * @apiParam (Body) {String[]} [reminders] Array of reminders, each an object that must
  *                                         include: a UUID, startDate and time.
@@ -550,7 +563,7 @@ api.getTask = {
  * @apiParam (Body) {String} [notes] Extra notes
  * @apiParam (Body) {Date} [date] Due date to be shown in task list. Only valid for type "todo."
  * @apiParam (Body) {Number="0.1","1","1.5","2"} [priority=1] Difficulty, options are 0.1, 1,
- *                                                            1.5, 2; eqivalent of Trivial,
+ *                                                            1.5, 2; equivalent of Trivial,
  *                                                            Easy, Medium, Hard.
  * @apiParam (Body) {String[]} [reminders] Array of reminders, each an object that must include:
  *                                         a UUID, startDate and time.
@@ -871,6 +884,7 @@ api.moveTask = {
  * @apiParam (Path) {String} taskId The task _id or alias
  *
  * @apiParam (Body) {String} text The text of the checklist item
+ * @apiParam (Body) {Boolean} [completed=false] Whether the checklist item is checked off.
  *
  * @apiParamExample {json} Example body data:
  * {"text":"Do this subtask"}
@@ -986,10 +1000,11 @@ api.scoreCheckListItem = {
  * @apiParam (Path) {String} taskId The task _id or alias
  * @apiParam (Path) {UUID} itemId The checklist item _id
  *
- * @apiParam (body) {String} text The text that will replace the current checkitem text.
+ * @apiParam (Body) {String} text The replacement text for the current checklist item.
+ * @apiParam (Body) {Boolean} [completed=false] Whether the checklist item is checked off.
  *
  * @apiParamExample {json} Example body:
- * {"text":"Czech 1"}
+ * {"text":"learn Czech", "completed":true}
  *
  * @apiSuccess {Object} data The updated task
  *

